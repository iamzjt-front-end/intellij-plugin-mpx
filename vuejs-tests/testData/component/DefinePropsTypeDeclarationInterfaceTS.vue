<script setup lang="ts">
// TODO Props should not be recognized as a component
interface Props {
  optional?: string,
  required: string,
}

defineProps<Props>();
</script>

<template>
  {{optional}}
  {{required}}
</template>
