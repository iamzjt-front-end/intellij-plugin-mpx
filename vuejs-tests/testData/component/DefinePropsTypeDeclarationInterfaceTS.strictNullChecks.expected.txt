{
 class: "VueSourceComponent",
 components: {
  DefinePropsTypeDeclarationInterfaceTS: <recursive value of class VueSourceComponent>,
  Props: {
   class: "VueLocallyDefinedRegularComponent",
   defaultName: "Props",
   data: [
    {
     class: "VueDecoratedDataProperty",
     name: "optional",
     jsType: "string",
    },
    {
     class: "VueDecoratedDataProperty",
     name: "required",
     jsType: "string",
    },
   ],
  },
 },
 props: [
  {
   class: "VueScriptSetupInputProperty",
   name: "optional",
   jsType: "string | undefined",
   required: false,
  },
  {
   class: "VueScriptSetupInputProperty",
   name: "required",
   jsType: "string",
   required: true,
  },
 ],
}
